starter_code/public/javascript/index.js

```diff
@@ -1,23 +1,105 @@
-  
-  document.getElementById('edit-character-form').onsubmit = function(){
-            
-  }
-  
-  document.getElementById('new-character-form').onsubmit = function(){
-                
-  }
-})
+  document.getElementById("fetch-one").onclick = function() {
+    charactersAPI.getOneRegister($("#search-input").prop("value")).then(character => {
+      // $("#character-container").html(`<pre>${JSON.stringify(characters, null, 2)}</pre>`);
+      $("#character-container").html(`
+        <div class="character-info" id="character-info">
+        <div class="name">Character Name : ${character.name}</div>
+        <div class="occupation">Character Occupation : ${character.occupation}</div>
+        <div class="debt">Character Debt : ${character.debt}</div>
+        <div class="weapon">Character Weapon : ${character.weapon}</div>
+      </div>
+      `);
+    });
+  };
+
+  document.getElementById("delete-one").onclick = function() {
+    charactersAPI.deleteOneRegister($("#delete-input").prop("value"));
+    //.then(character => {
+    // $("#character-container").html(`<pre>${JSON.stringify(characters, null, 2)}</pre>`);
+    //   $("#character-container").html(`
+    //     <div class="character-info" id="character-info">
+    //     <div class="name">Character Name : ${character.name}</div>
+    //     <div class="occupation">Character Occupation : ${character.occupation}</div>
+    //     <div class="debt">Character Debt : ${character.debt}</div>
+    //     <div class="weapon">Character Weapon : ${character.weapon}</div>
+    //   </div>
+    //   `);
+    // });
+  };
+
+  document.getElementById("edit-character-form").onsubmit = function() {
+    event.preventDefault();
+    if ($("#chr-id").val() !== "") {
+      if ($("#edited-name").val() !== "") {
+        charactersAPI
+          .updateOneRegister($("#chr-id").val(), {
+            name: $("#edited-name").val()
+          })
+          .then(character => {
+            console.log("Name updated");
+          });
+      }
+      if ($("#edited-occupation").val() !== "") {
+        charactersAPI
+          .updateOneRegister($("#chr-id").val(), {
+            occupation: $("#edited-occupation").val()
+          })
+          .then(character => {
+            console.log("occupation updated");
+          });
+      }
+      if ($("#edited-weapon").val() !== "") {
+        charactersAPI
+          .updateOneRegister($("#chr-id").val(), {
+            weapon: $("#edited-weapon").val()
+          })
+          .then(character => {
+            console.log("weapon updated");
+          });
+      }
+      if ($("#edited-debt").val() !== "") {
+        charactersAPI
+          .updateOneRegister($("#chr-id").val(), {
+            debt: $("#edited-debt").is(":checked")
+          })
+          .then(character => {
+            console.log("dept updated");
+          });
+      }
+    }
+  };
+
+  document.getElementById("new-character-form").onsubmit = function() {
+    event.preventDefault();
```

Comment:
Same here regarding the event. Another common convention is to name it `e`:
`document.getElementById("new-character-form").onsubmit = function(e) {
 e.preventDefault(); 
...
`